spec/datadog/profiling/http_transport_spec.rb

```diff
@@ -291,23 +292,28 @@
 
         expect(request.header).to include(
           'content-type' => [%r{^multipart/form-data; boundary=(.+)}],
+          'dd-evp-origin' => ['dd-trace-rb'],
+          'dd-evp-origin-version' => [DDTrace::VERSION::STRING],
         )
 
         # check body
         boundary = request['content-type'][%r{^multipart/form-data; boundary=(.+)}, 1]
         body = WEBrick::HTTPUtils.parse_form_data(StringIO.new(request.body), boundary)
+        event_data = JSON.parse(body.fetch('event'))
 
-        expect(body).to include(
-          'version' => '3',
-          'family' => 'ruby',
+        expect(event_data).to eq(
+          'attachments' => [pprof_file_name, code_provenance_file_name],
+          'tags_profiler' => 'tag_a:value_a,tag_b:value_b',
           'start' => start_timestamp,
           'end' => end_timestamp,
-          "data[#{pprof_file_name}]" => pprof_data,
-          "data[#{code_provenance_file_name}]" => code_provenance_data,
+          'family' => 'ruby',
+          'version' => '4',
         )
 
-        tags = body['tags[]'].list
-        expect(tags).to contain_exactly('tag_a:value_a', 'tag_b:value_b')
+        require 'extlz4' # Lazily required, to avoid trying to load it on JRuby
+
+        expect(LZ4.decode(body.fetch(pprof_file_name))).to eq pprof_data
+        expect(LZ4.decode(body.fetch(code_provenance_file_name))).to eq code_provenance_data
```

Comment:
I can't find to understand where this LZ4 compression is happening now.
I can only assume it now lives in libdatadog, but I can't find any hints in non-tests that

Reply:
Yes. The compression now entirely lives inside libdatadog, and is transparent. When calling `ddog_ProfileExporter_send` the byte arrays it receives with the data as compressed and sent to the backend.

> I can only assume it now lives in libdatadog, but I can't find any hints outside of this file of any LZ4 compression (or any form of compression) being done

You're right. And in hindsight maybe it's a bit too cryptic on the test.

I'll push a commit to move these tests to a specific `it 'reports the payload as lz4-compressed files, that get compressed by libdatadog'` so that this fact is clearly called out and documented in the code.

Reply:
Fixed in c8920081cb84180a8f824d34cb7532467353ce01